forge/routes/api/projectActions.js

```diff
@@ -108,4 +108,36 @@ module.exports = async function (app) {
             reply.code(500).send({ error: err.toString() })
         }
     })
+
+    app.post('/rollback', async (request, reply) => {
+        let restartProject = false
+        try {
+            // get (and check) snapshot is valid / owned by project before any actions
+            const snapshot = await app.db.models.ProjectSnapshot.byId(request.body.snapshotId)
```

Comment:
```suggestion
            const snapshot = await app.db.models.ProjectSnapshot.byId(request.body.snapshot)
```

For the external API, the property should be `snapshot` rather than `snapshotId`. This is more consistent with other places we pass object ids over the api.